FALCON-2039 Move falcon post processing to falcon server and remove p…

PraveenAdlakha opened the pull request:
…ost processing action from falcon workflow

PraveenAdlakha commented:
RETENTION:

$ hadoop dfs -text /projects/ivory/staging/falcon/workflows/feed/ProcessInstanceRerunTest1/_/_/workflow.xml
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<workflow-app xmlns="uri:oozie:workflow:0.3" name="FALCON_FEED_RETENTION_ProcessInstanceRerunTest1">
    <start to="eviction"/>
    <action name="eviction">
        <java>
            <job-tracker>${jobTracker}</job-tracker>
            <name-node>${nameNode}</name-node>
            <configuration>
                <property>
                    <name>mapred.job.queue.name</name>
                    <value>${queueName}</value>
                </property>
                <property>
                    <name>oozie.launcher.mapred.job.priority</name>
                    <value>${jobPriority}</value>
                </property>
                <property>
                    <name>oozie.action.sharelib.for.java</name>
                    <value>hcatalog,hive</value>
                </property>
                <property>
                    <name>oozie.launcher.oozie.libpath</name>
                    <value>${wf:conf(&quot;falcon.libpath&quot;)}</value>
                </property>
            </configuration>
            <main-class>org.apache.falcon.retention.FeedEvictor</main-class>
            <arg>-feedBasePath</arg>
            <arg>${feedDataPath}</arg>
            <arg>-falconFeedStorageType</arg>
            <arg>${falconFeedStorageType}</arg>
            <arg>-retentionType</arg>
            <arg>instance</arg>
            <arg>-retentionLimit</arg>
            <arg>${limit}</arg>
            <arg>-timeZone</arg>
            <arg>${timeZone}</arg>
            <arg>-frequency</arg>
            <arg>${frequency}</arg>
            <arg>-logFile</arg>
            <arg>${logDir}/job-${nominalTime}/${wf:run()}/evicted-instancePaths.csv</arg>
        </java>
        <ok to="end"/>
        <error to="fail"/>
    </action>
    <kill name="fail">
        <message>Workflow failed, error message[${wf:errorMessage(wf:lastErrorNode())}]</message>
    </kill>
    <end name="end"/>
</workflow-app>

RETENTION:

$ hadoop dfs -text /projects/ivory/staging/falcon/workflows/feed/ProcessInstanceRerunTest2/c14772932bba61d455a6eee640ae9dda_1469105899191/RETENTION/workflow.xml
DEPRECATED: Use of this script to execute hdfs command is deprecated.
Instead use the hdfs command for it.

/usr/lib/jvm/default-java//bin/java
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<workflow-app xmlns="uri:oozie:workflow:0.3" name="FALCON_FEED_RETENTION_ProcessInstanceRerunTest2">
    <start to="eviction"/>
    <action name="eviction">
        <java>
            <job-tracker>${jobTracker}</job-tracker>
            <name-node>${nameNode}</name-node>
            <configuration>
                <property>
                    <name>mapred.job.queue.name</name>
                    <value>${queueName}</value>
                </property>
                <property>
                    <name>oozie.launcher.mapred.job.priority</name>
                    <value>${jobPriority}</value>
                </property>
                <property>
                    <name>oozie.action.sharelib.for.java</name>
                    <value>hcatalog,hive</value>
                </property>
                <property>
                    <name>oozie.launcher.oozie.libpath</name>
                    <value>${wf:conf(&quot;falcon.libpath&quot;)}</value>
                </property>
            </configuration>
            <main-class>org.apache.falcon.retention.FeedEvictor</main-class>
            <arg>-feedBasePath</arg>
            <arg>${feedDataPath}</arg>
            <arg>-falconFeedStorageType</arg>
            <arg>${falconFeedStorageType}</arg>
            <arg>-retentionType</arg>
            <arg>instance</arg>
            <arg>-retentionLimit</arg>
            <arg>${limit}</arg>
            <arg>-timeZone</arg>
            <arg>${timeZone}</arg>
            <arg>-frequency</arg>
            <arg>${frequency}</arg>
            <arg>-logFile</arg>
            <arg>${logDir}/job-${nominalTime}/${wf:run()}/evicted-instancePaths.csv</arg>
        </java>
        <ok to="end"/>
        <error to="fail"/>
    </action>
    <kill name="fail">
        <message>Workflow failed, error message[${wf:errorMessage(wf:lastErrorNode())}]</message>
    </kill>
    <end name="end"/>
</workflow-app>

PROCESS:

$ hadoop dfs -text /projects/ivory/staging/falcon/workflows/process/ProcessInstanceRerunTest-agregator-28/_/_/workflow.xml
DEPRECATED: Use of this script to execute hdfs command is deprecated.
Instead use the hdfs command for it.

/usr/lib/jvm/default-java//bin/java
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<workflow-app xmlns="uri:oozie:workflow:0.3" name="FALCON_PROCESS_DEFAULT_ProcessInstanceRerunTest-agregator-28">
    <start to="user-action"/>
    <action name="user-action">
        <sub-workflow>
            <app-path>${nameNode}/user/dataqa/pig</app-path>
            <propagate-configuration/>
        </sub-workflow>
        <ok to="end"/>
        <error to="fail"/>
    </action>
    <kill name="fail">
        <message>Workflow failed, error message[${wf:errorMessage(wf:lastErrorNode())}]</message>
    </kill>
    <end name="end"/>
</workflow-app>

PraveenAdlakha committed:
FALCON-2039 Move falcon post processing to falcon server and remove post processing action from falcon workflow

PraveenAdlakha commented:
Lifecycle workflow:
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<workflow-app xmlns="uri:oozie:workflow:0.3" name="FALCON_FEED_RETENTION_ProcessInstanceRerunTest5">
    <start to="eviction"/>
    <action name="eviction">
        <java>
            <job-tracker>${jobTracker}</job-tracker>
            <name-node>${nameNode}</name-node>
            <configuration>
                <property>
                    <name>mapred.job.queue.name</name>
                    <value>${queueName}</value>
                </property>
                <property>
                    <name>oozie.launcher.mapred.job.priority</name>
                    <value>${jobPriority}</value>
                </property>
                <property>
                    <name>oozie.action.sharelib.for.java</name>
                    <value>hcatalog,hive</value>
                </property>
                <property>
                    <name>oozie.launcher.oozie.libpath</name>
                    <value>${wf:conf(&quot;falcon.libpath&quot;)}</value>
                </property>
            </configuration>
            <main-class>org.apache.falcon.retention.FeedEvictor</main-class>
            <arg>-feedBasePath</arg>
            <arg>${feedDataPath}</arg>
            <arg>-falconFeedStorageType</arg>
            <arg>${falconFeedStorageType}</arg>
            <arg>-retentionType</arg>
            <arg>instance</arg>
            <arg>-retentionLimit</arg>
            <arg>${limit}</arg>
            <arg>-timeZone</arg>
            <arg>${timeZone}</arg>
            <arg>-frequency</arg>
            <arg>${frequency}</arg>
            <arg>-logFile</arg>
            <arg>${logDir}/job-${nominalTime}/${wf:run()}/evicted-instancePaths.csv</arg>
        </java>
        <ok to="end"/>
        <error to="fail"/>
    </action>
    <kill name="fail">
        <message>Workflow failed, error message[${wf:errorMessage(wf:lastErrorNode())}]</message>
    </kill>
    <end name="end"/>
</workflow-app>

PraveenAdlakha committed:
check added in service

ajayyadava commented on common/src/main/resources/runtime.properties:
```diff
@@ -51,4 +51,7 @@
+######### Proxyuser Configuration End #########
+
+# Property to remove postProcessing
+#*.falcon.disable.postprocessing=true
```
default should be false for backward compatibility, right?

ajayyadava replied:
Rename to falcon.postprocessing.disable=false

ajayyadava commented on common/src/main/resources/startup.properties:
```diff
@@ -83,6 +83,7 @@
 
 ##### Workflow Job Execution Completion listeners #####
 *.workflow.execution.listeners=
+#org.apache.falcon.service.LogMoverService
```
Relate the 2 properties with each other through comments so that the users know that both should be switched together.

ajayyadava commented on oozie/src/main/java/org/apache/falcon/oozie/OozieOrchestrationWorkflowBuilder.java:
```diff
@@ -79,6 +79,8 @@
  */
 public abstract class OozieOrchestrationWorkflowBuilder<T extends Entity> extends OozieEntityBuilder<T> {
     public static final String HIVE_CREDENTIAL_NAME = "falconHiveAuth";
+    public static final String DISABLE_POSTPROCESSING = RuntimeProperties.get().
```
POSTPROCESSING_ENABLED

ajayyadava commented on oozie/src/main/java/org/apache/falcon/service/LogMoverService.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    @Override
+    public void onStart(WorkflowExecutionContext context) throws FalconException{
```
Remove start, suspend.

PraveenAdlakha committed:
minor fixes done

PraveenAdlakha committed:
Startup properties

PraveenAdlakha committed:
Agebased boolean check removed

PraveenAdlakha committed:
multithread added

PraveenAdlakha committed:
documentation added

PraveenAdlakha committed:
fixed test cases

PraveenAdlakha committed:
comments addressed

PraveenAdlakha committed:
process removed

Commit:
FALCON-2072 Hive2 URLs in Falcon should allow additional configuration elements in the URL

Author: Sowmya Ramesh <sramesh@hortonworks.com>

Reviewers: "Venkat  Ranganathan <venkat@hortonworks.com>, Balu Vellanki <balu@apache.org>"

Closes #250 from sowmyaramesh/FALCON-2072

Commit:
FALCON-1944 Ability to provide additional DistCP options for mirroring extensions and feed replication

Author: Sowmya Ramesh <sramesh@hortonworks.com>

Reviewers: "Venkat  Ranganathan <venkat@hortonworks.com>, Balu Vellanki <balu@apache.org>"

Closes #251 from sowmyaramesh/FALCON-1944
